src/io/file_descriptor.cr

```diff
@@ -149,7 +154,8 @@ class IO::FileDescriptor < IO
 
   def reopen(other : IO::FileDescriptor)
     system_reopen(other)
-
+    # You shouldn't be able to reopen a closed FD
+    @closed = false
```

Comment:
This should be put in `system_reopen`, just after the `dup` call, so there's not a race between the reschedule and setting `@closed = false`.